<a href="https://colab.research.google.com/github/trinhvanypro1/-DataAnalystDeepLearning/blob/main/Lab6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***BÀI 1:*** 
Dữ liệu sử dụng trong lab này là tập dữ liệu về lượng nhựa, nicotine, CO trong
thuốc lá cỡ king

In [27]:
import pandas as pd
data = pd.read_csv('https://raw.githubusercontent.com/huynhhoc/DataAnalystDeepLearning/main/Data/04_CIGARET.csv')
data.head(10)

TypeError: ignored

In [28]:
# Ex1:
import statsmodels.formula.api as smf
res = smf.ols('KgNic~KgTar+KgCO',data).fit()
print(res.summary())

                            OLS Regression Results                            
Dep. Variable:                  KgNic   R-squared:                       0.099
Model:                            OLS   Adj. R-squared:                  0.017
Method:                 Least Squares   F-statistic:                     1.211
Date:                Fri, 29 Oct 2021   Prob (F-statistic):              0.317
Time:                        03:45:32   Log-Likelihood:                 2.7726
No. Observations:                  25   AIC:                            0.4548
Df Residuals:                      22   BIC:                             4.111
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.5937      0.793      2.010      0.0

**Nhận xét:**

  - *Kiểm định giả thuyết hệ số hồi quy*:
    + Hệ số hồi quy của biến KgTar = 0.0231 với P-value = 0,153 > 0,05. Do P-value > 0.05, điều này có ý nghĩa tác động của biến KgTar lên KgNic không có ý nghĩa thống kê.

    + Hệ số hồi quy của biến KgCO = -0.0525 với P-value = 0.340 > 0.05. Do P-value > 0.05 nên, điều này có ý nghĩa tác động của biến KgCO ên KgNic không có ý nghĩa thống kê.

  - Kiểm định mô hình (Kiểm định F):

     + Giá trị F = 1.211 với P-values = 0.317 > 0.05. Do P-value > 0.05 nên mô hình này cũng không có ý nghĩa thống kê.

In [29]:
#Ex 2:
resMn = smf.ols('MnNic~MnTar+MnCO',data).fit()
print(resMn.summary())

                            OLS Regression Results                            
Dep. Variable:                  MnNic   R-squared:                       0.915
Model:                            OLS   Adj. R-squared:                  0.908
Method:                 Least Squares   F-statistic:                     118.8
Date:                Fri, 29 Oct 2021   Prob (F-statistic):           1.62e-12
Time:                        03:45:32   Log-Likelihood:                 31.318
No. Observations:                  25   AIC:                            -56.64
Df Residuals:                      22   BIC:                            -52.98
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.2507      0.057      4.399      0.0

***Nhận xét:***

  - *Kiểm định giả thuyết của hệ số hổi quy:*
    
    + Hệ số hồi quy của MnTar = 0,1007 với P-value = 0.000 < 0.05. Điều này có ý nghĩa tác động của biến MnTar lên MnNic có ý nghĩa thống kê.

    + Hệ số hồi quy của MnCO = -0,0454 với P-value = 0.000 < 0.05. Điều này có ý nghĩa tác động của biến MnCO lên MnNic có ý nghĩa thống kê.

  - *Kiểm định mô hình (Kiểm định F)*:
    + Giá trị F =0.015 với P-value = 1.62e-12 < 0.05. Do P-value < 0.05 nên mô hình này có ý nghĩa thống kê

# ***BÀI 2:***
Dữ liệu sử dụng trong lab này là dữ liệu về kích thước giáp cua.

In [35]:
import pandas as pd
import numpy as np


In [36]:
df = pd.read_csv('https://raw.githubusercontent.com/huynhhoc/DataAnalystDeepLearning/main/Data/crabs.txt',sep='\s+')

TypeError: ignored

In [34]:
df.head(10)

AttributeError: ignored